module/extension/Director/src/director/challenge_priority.cpp

```diff
@@ -75,7 +83,7 @@ namespace module::extension {
 
                 // Add this new parent we found and set t so we can loop up further
                 t = group.active_task;
```

Comment:
Is this supposed to be overwriting the iterator variable here? It gives me bad juju but I guess the goal is to skip straight to the active task if possible?

Reply:
It is acting like an iterator variable, but really it's just a pointer (a shared_ptr in this case).
This part of the algorithm is just moving up the tree to find each direct ancestor